gr-digital/lib/crc16_async_bb_impl.cc

```diff
@@ -0,0 +1,105 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2021 Cesar Martinez.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * 
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "crc16_async_bb_impl.h"
+#include <gnuradio/io_signature.h>
+#include <volk/volk.h>
+
+namespace gr {
+namespace digital {
+
+crc16_async_bb::sptr crc16_async_bb::make(bool check)
+{
+    return gnuradio::make_block_sptr<crc16_async_bb_impl>(check);
+}
+
+crc16_async_bb_impl::crc16_async_bb_impl(bool check)
+    : block("crc16_async_bb", io_signature::make(0, 0, 0), io_signature::make(0, 0, 0)),
+      d_npass(0),
+      d_nfail(0)
+{
+    d_in_port = pmt::mp("in");
+    d_out_port = pmt::mp("out");
```

Comment:
I think the code base is scattered of examples like this, and also many examples where `pmt::mp("out")` is repeated several times throughout the block implementation instead of being stored in a class member as here.

Perhaps an idea for improvement is to use [`msgport_names`](https://github.com/gnuradio/gnuradio/blob/4d306e1f98eb60cf62ca8f0aa59e3939e7826e84/gnuradio-runtime/include/gnuradio/pdu.h#L19), which contains standard PDU port names. However `"in"`, and `"out"` are not included there, so we'd like to add them first. Maybe it's worth discussing how to improve this in a separate issue?

For this PR I think this is OK, since I guess it's copied over from the other CRC block.